### PR TITLE
PIM-7629-bis

### PR DESCRIPTION
**Pull request by momoss**
**Description (for Contributor and Core Developer)**
Example:
Model -> in category "headphone"
Variant -> parent "model" -> herited of his parent

When you search on "Audio and video" subcategory, (parent of Headphone), both model & variant appear.
 
image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | OK
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Resources/config/query_builders.yml [hunk 1]
```diff
@@ -283,6 +283,15 @@ services:
             - { name: 'pim_catalog.elasticsearch.query.product_model_filter', priority: 30 }
             - { name: 'pim_catalog.elasticsearch.query.product_and_product_model_filter', priority: 30 }
 
+    pim_catalog.query.elasticsearch.filter.category_of_ancestors:
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Resources/config/query_builders.yml]: indent
This service should not be exposed as a user filter of the PQB

**Comment by SamirBoulil** [comment]
It would have been helpful if you wrote a "Proposed solution" section in the description's PR, for the next time ;)

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Field/AggregateFilter.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+use Akeneo\Component\Classification\Repository\CategoryRepositoryInterface;
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+use Pim\Component\Catalog\Exception\InvalidOperatorException;
+use Pim\Component\Catalog\Query\Filter\FieldFilterHelper;
+use Pim\Component\Catalog\Query\Filter\FieldFilterInterface;
+use Pim\Component\Catalog\Query\Filter\Operators;
+
+/**
+ * @author    Philippe Mossière <philippe.mossiere@akeneo.com>
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Field/AggregateFilter.php]: Explain the purpose here

**Comment by momoss** [comment]
look at https://github.com/akeneo/pim-community-dev/pull/8875

**Pull request closed without merging**